queue_services/entity-filer/tests/unit/test_worker/test_correction_ben.py

```diff
@@ -0,0 +1,709 @@
+
+    filing = copy.deepcopy(filing_template)
+
+    corrected_filing_id = factory_completed_filing(business, BEN_CORRECTION_APPLICATION).id
+    filing['filing']['correction']['correctedFilingId'] = corrected_filing_id
+
+    if test_name == 'name_change':
+        filing['filing']['correction']['nameRequest']['legalName'] = new_legal_name
+        filing['filing']['business']['legalName'] = new_legal_name
+    else:
+        del filing['filing']['correction']['nameRequest']
+
+    payment_id = str(random.SystemRandom().getrandbits(0x58))
+
+    filing_id = (create_filing(payment_id, filing, business_id=business_id)).id
+    filing_msg = {'filing': {'id': filing_id}}
+
+    # mock out the email sender and event publishing
+    mocker.patch('entity_filer.worker.publish_email_message', return_value=None)
+    mocker.patch('entity_filer.worker.publish_event', return_value=None)
+    mocker.patch('entity_filer.filing_processors.filing_components.name_request.consume_nr', return_value=None)
+    mocker.patch('entity_filer.filing_processors.filing_components.business_profile.update_business_profile',
+                 return_value=None)
+    mocker.patch('legal_api.services.bootstrap.AccountService.update_entity', return_value=None)
+
+    # Test
+    with patch.object(NaicsService, 'find_by_code', return_value=naics_response):
+        await process_filing(filing_msg, app)
+
+    # Check outcome
+    final_filing = Filing.find_by_id(filing_id)
+    correction = final_filing.meta_data.get('correction', {})
+    business = Business.find_by_internal_id(business_id)
+
+    if new_legal_name:
+        assert business.legal_name == new_legal_name
+        assert correction.get('toLegalName') == new_legal_name
+        assert correction.get('fromLegalName') == legal_name
+    else:
+        assert business.legal_name == legal_name
+        assert correction.get('toLegalName') is None
+        assert correction.get('fromLegalName') is None
+
+    corrected_filing = Filing.find_by_id(corrected_filing_id)
+    filing_comments = final_filing.comments.all()
+    assert len(filing_comments) == 1
+    assert filing_comments[0].comment == filing['filing']['correction']['comment']
+    assert len(corrected_filing.comments.all()) == 1
+    assert business.start_date.isoformat() == '2022-01-01T08:00:00+00:00'
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    'test_name, legal_type, legal_name, filing_template',
+    [
+        ('ben_address_change', 'BEN', 'Test Firm', BEN_CORRECTION)
+    ]
+)
+async def test_correction_business_address(app, session, mocker, test_name, legal_type, legal_name,
+                                                       filing_template):
+    """Assert the worker process calls the business address change correctly."""
+    identifier = 'BC1234567'
+    business = create_entity(identifier, legal_type, legal_name)
+    business.save()
+    business_id = business.id
+
+    office = create_office(business, 'registeredOffice')
```

Comment:
sonarcloud doesn't like this line

Reply:
I'm not sure why sonacloud doesn't like this line, because this is the same line as the following lines:

https://github.com/bcgov/lear/blob/802ab7216d72d14e15d2636c714d7122545c7a39/queue_services/entity-filer/tests/unit/test_worker/test_correction_firms.py#L153

https://github.com/bcgov/lear/blob/802ab7216d72d14e15d2636c714d7122545c7a39/queue_services/entity-filer/tests/unit/test_worker/test_change_of_registration.py#L135

Reply:
ok, yeah i can't see what the problem is there.  let's leave it